client/rest-high-level/src/test/java/org/elasticsearch/client/analytics/InferenceAggIT.java

```diff
@@ -0,0 +1,127 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.analytics;
+
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.client.ESRestHighLevelClientTestCase;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.indices.CreateIndexRequest;
+import org.elasticsearch.client.ml.PutTrainedModelRequest;
+import org.elasticsearch.client.ml.inference.TrainedModelConfig;
+import org.elasticsearch.client.ml.inference.TrainedModelDefinition;
+import org.elasticsearch.client.ml.inference.TrainedModelInput;
+import org.elasticsearch.client.ml.inference.trainedmodel.RegressionConfig;
+import org.elasticsearch.client.ml.inference.trainedmodel.tree.Tree;
+import org.elasticsearch.client.ml.inference.trainedmodel.tree.TreeNode;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.search.aggregations.bucket.terms.ParsedTerms;
+import org.elasticsearch.search.aggregations.bucket.terms.Terms;
+import org.elasticsearch.search.aggregations.bucket.terms.TermsAggregationBuilder;
+import org.elasticsearch.search.aggregations.metrics.AvgAggregationBuilder;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.closeTo;
+import static org.hamcrest.Matchers.equalTo;
+
+public class InferenceAggIT extends ESRestHighLevelClientTestCase {
+
+    public void testInferenceAgg() throws IOException {
+
+        // create a very simple decision tree with a root node and 2 leaves
+        List<String> featureNames = Collections.singletonList("cost");
+        Tree.Builder builder = Tree.builder();
+        builder.setFeatureNames(featureNames);
+        TreeNode.Builder root = builder.addJunction(0, 0, true, 1.0);
+        int leftChild = root.getLeftChild();
+        int rightChild = root.getRightChild();
+        builder.addLeaf(leftChild, 10.0);
+        builder.addLeaf(rightChild, 20.0);
+
+        final String modelId = "simple_regression";
+        putTrainedModel(modelId, featureNames, builder.build());
+
+        final String index = "inference-test-data";
+        indexData(index);
+
+        TermsAggregationBuilder termsAgg = new TermsAggregationBuilder("fruit_type").field("fruit");
+        AvgAggregationBuilder avgAgg = new AvgAggregationBuilder("avg_cost").field("cost");
+        termsAgg.subAggregation(avgAgg);
+
+        Map<String, String> bucketPaths = new HashMap<>();
+        bucketPaths.put("cost", "avg_cost");
+        InferencePipelineAggregationBuilder inferenceAgg = new InferencePipelineAggregationBuilder("infer", modelId,  bucketPaths);
```

Comment:
I think it would be cool to also have a test that adds at least one option. Possibly `feature_importance` calculation.